src/Diagnostics.DataProviders/GeoMasterUtility.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Web;
+
+namespace Diagnostics.DataProviders
+{
+    public class GeoMasterConstants
+    {
+        public const string August2016Version = "2016-08-01";
+        public const string ProductionSlot = "Production";
+        public const string SlotNameTemplateParameter = "/{slot}";
+        public const string SlotsResource = "/slots";
+        public const string NameTemplateParamater = "/{name}";
+        public const string SitesResource = "sites";
+        public const string Functions = "functions";
+        public const string ResourceProviderName = "Microsoft.Web";
+        public const string NetworkResourceProviderName = "Microsoft.Network";
+        public const string ResourceProviderDisplayName = "Microsoft Web Apps";
+        public const string ResourceGroupSegment = "resourceGroups";
+        public const string ResourceGroupRoot = "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}";
+        public const string ResourceRoot =
+                "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/" + ResourceProviderName;
+    }
+    public class DataSourceCancelledException : Exception
```

Comment:
Should this be in a seperate file? Is it supposed to be used across any data provider or just geomaster?

If you leave it here please add a empty line between this and the class above

Reply:
I would like to understand better on how we are handling cancellations and where should we be logging exceptions. Then we can decide if this should be same or different file.

Reply:
Actually, can we make this a in a different file like CustomExceptions.cs or something. Even before tackling the logging story this exception would be useful across all the data providers.

Reply:
Created a common folder and moved this file over there...